actors/builtin/multisig/multisig_test.go

```diff
@@ -76,6 +76,114 @@ func TestConstruction(t *testing.T) {
+
+	t.Run("happy path full vesting", func(t *testing.T) {
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, 2, unlockDuration, []addr.Address{anne, bob, charlie}...)
+
+		// anne proposes that darlene receives `multisgiInitialBalance` FIL.
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.propose(rt, darlene, abi.NewTokenAmount(multisigInitialBalance), builtin.MethodSend, nilParams)
+		rt.Verify()
+
+		// Advance the epoch s.t. all funds are unlocked.
+		rt.SetEpoch(0 + unlockDuration)
+		// bob approves annes transaction
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		// expect darlene to receive the transaction proposed by anne.
+		rt.ExpectSend(darlene, builtin.MethodSend, nilParams, abi.NewTokenAmount(multisigInitialBalance), nil, exitcode.Ok)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.approve(rt, 0)
+		rt.Verify()
+
+	})
+
+	t.Run("partial vesting", func(t *testing.T) {
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, 2, 10, []addr.Address{anne, bob, charlie}...)
+
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.propose(rt, darlene, abi.NewTokenAmount(multisigInitialBalance/2), builtin.MethodSend, nilParams)
+		rt.Verify()
+
+		rt.SetEpoch(0 + unlockDuration/2)
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		rt.ExpectSend(darlene, builtin.MethodSend, nilParams, abi.NewTokenAmount(multisigInitialBalance/2), nil, exitcode.Ok)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.approve(rt, 0)
+		rt.Verify()
+
+	})
+
+	t.Run("pass proposed transaction on approval after epoch passes", func(t *testing.T) {
+		rt := builder.Build()
+
+		actor.constructAndVerify(rt, 1, unlockDuration, []addr.Address{anne, bob, charlie}...)
+
+		// this propose will fail since it would send more than the required locked balance and num approvals == 1
+		rt.SetCaller(anne, builtin.AccountActorCodeID)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		rt.ExpectAbort(exitcode.ErrInsufficientFunds, func() {
+			actor.propose(rt, darlene, abi.NewTokenAmount(multisigInitialBalance), builtin.MethodSend, nilParams)
+
+		})
+		rt.Verify()
+
+		// move the epoch ahead such that all FIL is unlocked.
+		rt.SetEpoch(unlockDuration)
+		// bob approves the transaction proposed by anne to send darlene FIL.
+		rt.SetCaller(bob, builtin.AccountActorCodeID)
+		// we expect a message send to darlene containing the method and value proposed by anne.
+		rt.ExpectSend(darlene, builtin.MethodSend, nilParams, abi.NewTokenAmount(multisigInitialBalance), nil, exitcode.Ok)
+		rt.ExpectValidateCallerType(builtin.AccountActorCodeID, builtin.MultisigActorCodeID)
+		actor.approve(rt, 0)
```

Comment:
This exposes a flaw in our runtime mock. The aborted `propose` should not persist state.